mobile/Droidlet/Shared/Firebase.swift

```diff
@@ -0,0 +1,134 @@
+import ARKit
+import RealityKit
+import Firebase
+import FirebaseStorage
+import FirebaseAuth
+import Foundation
+import UIKit
+
+class Droidbase {
+    
+    private static let userCvPath = "human-object-recognition"
+    private let cvDbRef = Database.database().reference(withPath: userCvPath)
+    private let cvDbRef3D = Database.database().reference(withPath: "human-object-recognition3D")
+    private let storageRef = Storage.storage().reference()
+    
+    static let shared = Droidbase()
+    
+    func saveToCloud(_ arView: ARView?, _ name:String, _ image: UIImage) {
+        guard let data = image.jpegData(compressionQuality: 0.6) else {
+            fatalError("Can't save snaphot")
+        }
+        let metadata = StorageMetadata()
+        metadata.contentType = "image/jpg"
+        let timestamp = String(Date().timeIntervalSince1970)
+        //let mesh = serializedMap(sceneView: arView)
+        
+        let dataRef = storageRef.child("images/\(timestamp).jpg")
+        dataRef.putData(data, metadata: metadata)
+        cvDbRef.childByAutoId().setValue(["name": name, "snapshot": timestamp, "mesh": "mesh"])
+    }
+    
+    func save3DToCloud(_ arView: ARView?, _ name:String, url: URL?) {
+        guard let url = url else {
+            return
+        }
+
+        do {
+            let data = try Data(contentsOf: url)
+            let metadata = StorageMetadata()
+            metadata.contentType = "image/obj"
+            let timestamp = String(Date().timeIntervalSince1970)
+            
+            let dataRef = storageRef.child("images3d/\(timestamp).obj")
+            dataRef.putData(data, metadata: metadata)
+            cvDbRef3D.childByAutoId().setValue(["name": name, "snapshot": timestamp, "mesh": "mesh"])
+        } catch {
+            
+        }
+    }
+    
+    func serializedMap(sceneView: ARView) -> String {
+        let worldMap = sceneView.session.getCurrentWorldMap
+        do {
+            let data = try NSKeyedArchiver.archivedData(withRootObject: worldMap, requiringSecureCoding: true)
+            return data.base64EncodedString()
+        } catch {
+            fatalError("Can't save map: \(error.localizedDescription)")
+        }
+    }
+    
+    func getObjectAPI(completion: ((UIImage) -> ())?) {
+        let semaphore = DispatchSemaphore (value: 0)
+        
+        var request = URLRequest(url: URL(string: "https://yss21031-default-rtdb.firebaseio.com/human-object-recognition.json")!,timeoutInterval: Double.infinity)
```

Comment:
instead of adding a hardcoded value for the URL here, let's make this a variable and add this as default value